lib/puppet/defaults.rb

```diff
@@ -419,7 +419,20 @@ def self.default_diffargs
     :data_binding_terminus => {
       :type    => :terminus,
       :default => "hiera",
-      :desc    => "Where to retrieve information about data.",
+      :desc    =>
+        "This setting has been deprecated. Use of any value other than 'hiera' should instead be configured
+         in a version 5 hiera.yaml. Until this setting is removed, it controls which data binding terminus
+         to use for global automatic data binding (across all environments). By default this value is 'hiera'.
+         A value of 'none' turns off the global binding.",
+      :call_hook => :on_initialize_and_write,
+      :hook => proc do |value|
+        if Puppet[:strict] != :off
+          s_val = value.to_s # because sometimes the value is a symbol
+          unless s_val == 'hiera' || s_val == 'none' || value == '' || value.nil?
+            Puppet.deprecation_warning "Setting 'data_binding_terminus' is deprecated. Convert custom terminus to hiera 5 API."
```

Comment:
A minor wording suggestion:
change
```
Convert custom terminus to hiera 5 API
```
to
```
Convert custom terminus to hiera.yaml API version 5
```

or something similar. Basically when I first read this I was confused and thought the suggestion was to upgrade to Hiera 5 - but Hiera 3 is current. Perhaps specifying that we're talking about version 5 of the API is more intuitive?